### PR TITLE
Scope app spec dependent targets when no dedup'ing targets.

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

File: lib/cocoapods/target/pod_target.rb [hunk 1]
```diff
@@ -117,15 +117,17 @@ def scoped(cache = {})
...
-              test_pod_target.scoped(cache).select { |pt| pt.target_definitions == [target_definition] }
-            end
-            [spec_name, scoped_test_pod_targets]
-          end]
+          target = PodTarget.new(sandbox, host_requires_frameworks, user_build_configurations, archs, platform, specs,
+                                 [target_definition], file_accessors, target_definition.label, :build_type => build_type)
+          target.dependent_targets = dependent_targets.flat_map { |pt| pt.scoped(cache) }.select do |pt|
+            pt.target_definitions == [target_definition]
+          end
+          target.test_dependent_targets_by_spec_name = scope_dependent_pod_targets(cache,
```
Comment (dnkoutso) [inline lib/cocoapods/target/pod_target.rb]: extracted method

**Comment by segiddins** [comment]
I think this'll conflict since I made a similar fix on master

**Comment by dnkoutso** [comment]
I'll resolve it.

File: lib/cocoapods/target/pod_target.rb [hunk 2]
```diff
@@ -117,15 +117,17 @@ def scoped(cache = {})
...
-          target.test_dependent_targets_by_spec_name = Hash[test_dependent_targets_by_spec_name.map do |spec_name, test_pod_targets|
-            scoped_test_pod_targets = test_pod_targets.flat_map do |test_pod_target|
-              test_pod_target.scoped(cache).select { |pt| pt.target_definitions == [target_definition] }
-            end
-            [spec_name, scoped_test_pod_targets]
-          end]
+          target = PodTarget.new(sandbox, host_requires_frameworks, user_build_configurations, archs, platform, specs,
+                                 [target_definition], file_accessors, target_definition.label, :build_type => build_type)
+          target.dependent_targets = dependent_targets.flat_map { |pt| pt.scoped(cache) }.select do |pt|
+            pt.target_definitions == [target_definition]
```
Comment (amorde) [inline lib/cocoapods/target/pod_target.rb]: super nit: this creates a new array in the loop, pulling it out would save some memory

File: lib/cocoapods/target/pod_target.rb [hunk 3]
```diff
@@ -764,6 +766,15 @@ def uses_modular_headers?(only_if_defines_modules = true)
 
     private
 
+    def scope_dependent_pod_targets(cache, dependent_targets, target_definition)
+      Hash[dependent_targets.map do |spec_name, pod_targets|
+        scoped_pod_targets = pod_targets.flat_map do |pod_target|
+          pod_target.scoped(cache).select { |pt| pt.target_definitions == [target_definition] }
```
Comment (amorde) [inline lib/cocoapods/target/pod_target.rb]: same as above

**Pull request merged**